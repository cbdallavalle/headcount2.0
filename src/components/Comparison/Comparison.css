.comparison {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  margin: 10px;
  padding: 10px;
  box-sizing: border-box;
  box-shadow: 2px 2px 2px 2px #e4e4e4;
  background-color: #f4f4f4;
  border-radius: 40px;
  height: 346px;
  width: 250px;
  text-align: center;
}

.displayAll {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: fit-content;
}

button {
  align-self: flex-start;
  margin: 10px;
  padding: 10px;
  box-sizing: border-box;
  box-shadow: 1px 1px 1px 1px #b0b0b0;
  background-color: #f4f4f4fa;
  border: none;
  border-radius: 15px;
  font-family: 'Bree Serif', serif;
  font-size: 1em;
}

button:hover {
  background-color: #dd4a0d;
  cursor: pointer;
}