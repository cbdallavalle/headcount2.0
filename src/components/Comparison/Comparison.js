import React from 'react';
import Card from '../Card/Card';

const Comparison = ({comparisonData, handleCompareCards, removeComparison}) => {
  
  return (
    <section className="displayAll">
      <Card 
        schoolData={ comparisonData[0] }
        handleCompareCards={ handleCompareCards }
        size='large'
      />

      <article className="comparison">
        <div>
          <h3>{ comparisonData[0].location }</h3>
          <h4>{ comparisonData[2][Object.keys(comparisonData[2])[0]] }</h4>
        </div>
        <div>
          <h3>Differential</h3>
          <h4 className='compare-total'>{ comparisonData[2].compared }</h4>
        </div>
        <div>
          <h3>{ comparisonData[1].location }</h3>
          <h4>{ comparisonData[2][Object.keys(comparisonData[2])[1]] }</h4>
        </div>
      </article>
      
      <Card 
        schoolData={ comparisonData[1] }
        handleCompareCards={ handleCompareCards }
        size='large'
      />
      <button onClick={ removeComparison }>Reset</button>
    </section>
  )
}

export default Comparison;