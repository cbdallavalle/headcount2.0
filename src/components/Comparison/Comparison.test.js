/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Comparison from './Comparison';

describe('Comparison', () => {
  let wrapper;
  let mockComparisonData;
  let mockRemoveComparison;
  
  beforeEach( () => {
    mockRemoveComparison = jest.fn();
    mockComparisonData = {
      school1: {
        data: {2004: 0.302, 2005: 0.267, 2006: 0.354, 2007: 0.392, 2008: 0.385,
          2009: 0.39, 2010: 0.436, 2011: 0.489, 2012: 0.479, 2013: 0.488, 2014: 0.49},
        dataType: "Percent",
        location: "ACADEMY 20"
      },
      school2: {
        data: {2004: 0, 2005: 0, 2006: 0, 2007: 0, 2008: 0.844, 2009: 0.96, 
          2010: 1, 2011: 1, 2012: 1, 2013: 1, 2014: 1},
        dataType: "Percent",
        location: "AKRON R-1"
      },
      comparison: {
        "ACADEMY 20": 0.407, "AKRON R-1": 0.619, compared: 0.658
      }
    }
    wrapper = shallow(<Comparison
      comparisonData={mockComparisonData}
      handleCompareCards={ () => {} }
      removeComparison={mockRemoveComparison}
                      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should execute removeComparison when the button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockRemoveComparison.mock.calls.length).toBe(1)
  })
})
