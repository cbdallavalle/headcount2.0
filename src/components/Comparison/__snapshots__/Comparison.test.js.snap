// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comparison
    comparisonData={
        Object {
            "comparison": Object {
              "ACADEMY 20": 0.407,
              "AKRON R-1": 0.619,
              "compared": 0.658,
            },
            "school1": Object {
              "data": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
              "dataType": "Percent",
              "location": "ACADEMY 20",
            },
            "school2": Object {
              "data": Object {
                "2004": 0,
                "2005": 0,
                "2006": 0,
                "2007": 0,
                "2008": 0.844,
                "2009": 0.96,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "dataType": "Percent",
              "location": "AKRON R-1",
            },
          }
    }
    handleCompareCards={[Function]}
    removeComparison={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          handleCompareCards={[Function]}
          schoolData={
                    Object {
                              "data": Object {
                                "2004": 0.302,
                                "2005": 0.267,
                                "2006": 0.354,
                                "2007": 0.392,
                                "2008": 0.385,
                                "2009": 0.39,
                                "2010": 0.436,
                                "2011": 0.489,
                                "2012": 0.479,
                                "2013": 0.488,
                                "2014": 0.49,
                              },
                              "dataType": "Percent",
                              "location": "ACADEMY 20",
                            }
          }
          size="large"
/>,
        <article
          className="comparison"
>
          <div>
                    <h3>
                              ACADEMY 20
                    </h3>
                    <h4>
                              0.407
                    </h4>
          </div>
          <div>
                    <h3>
                              Differential
                    </h3>
                    <h4
                              className="compare-total"
                    >
                              0.658
                    </h4>
          </div>
          <div>
                    <h3>
                              AKRON R-1
                    </h3>
                    <h4>
                              0.619
                    </h4>
          </div>
</article>,
        <Card
          handleCompareCards={[Function]}
          schoolData={
                    Object {
                              "data": Object {
                                "2004": 0,
                                "2005": 0,
                                "2006": 0,
                                "2007": 0,
                                "2008": 0.844,
                                "2009": 0.96,
                                "2010": 1,
                                "2011": 1,
                                "2012": 1,
                                "2013": 1,
                                "2014": 1,
                              },
                              "dataType": "Percent",
                              "location": "AKRON R-1",
                            }
          }
          size="large"
/>,
        <button
          onClick={[Function]}
>
          Reset
</button>,
      ],
      "className": "displayAll",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleCompareCards": [Function],
          "schoolData": Object {
            "data": Object {
              "2004": 0.302,
              "2005": 0.267,
              "2006": 0.354,
              "2007": 0.392,
              "2008": 0.385,
              "2009": 0.39,
              "2010": 0.436,
              "2011": 0.489,
              "2012": 0.479,
              "2013": 0.488,
              "2014": 0.49,
            },
            "dataType": "Percent",
            "location": "ACADEMY 20",
          },
          "size": "large",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h3>
                            ACADEMY 20
              </h3>
              <h4>
                            0.407
              </h4>
</div>,
            <div>
              <h3>
                            Differential
              </h3>
              <h4
                            className="compare-total"
              >
                            0.658
              </h4>
</div>,
            <div>
              <h3>
                            AKRON R-1
              </h3>
              <h4>
                            0.619
              </h4>
</div>,
          ],
          "className": "comparison",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  ACADEMY 20
</h3>,
                <h4>
                  0.407
</h4>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ACADEMY 20",
                },
                "ref": null,
                "rendered": "ACADEMY 20",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0.407,
                },
                "ref": null,
                "rendered": 0.407,
                "type": "h4",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Differential
</h3>,
                <h4
                  className="compare-total"
>
                  0.658
</h4>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Differential",
                },
                "ref": null,
                "rendered": "Differential",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0.658,
                  "className": "compare-total",
                },
                "ref": null,
                "rendered": 0.658,
                "type": "h4",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  AKRON R-1
</h3>,
                <h4>
                  0.619
</h4>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AKRON R-1",
                },
                "ref": null,
                "rendered": "AKRON R-1",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0.619,
                },
                "ref": null,
                "rendered": 0.619,
                "type": "h4",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleCompareCards": [Function],
          "schoolData": Object {
            "data": Object {
              "2004": 0,
              "2005": 0,
              "2006": 0,
              "2007": 0,
              "2008": 0.844,
              "2009": 0.96,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 1,
              "2014": 1,
            },
            "dataType": "Percent",
            "location": "AKRON R-1",
          },
          "size": "large",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Reset",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            handleCompareCards={[Function]}
            schoolData={
                        Object {
                                    "data": Object {
                                      "2004": 0.302,
                                      "2005": 0.267,
                                      "2006": 0.354,
                                      "2007": 0.392,
                                      "2008": 0.385,
                                      "2009": 0.39,
                                      "2010": 0.436,
                                      "2011": 0.489,
                                      "2012": 0.479,
                                      "2013": 0.488,
                                      "2014": 0.49,
                                    },
                                    "dataType": "Percent",
                                    "location": "ACADEMY 20",
                                  }
            }
            size="large"
/>,
          <article
            className="comparison"
>
            <div>
                        <h3>
                                    ACADEMY 20
                        </h3>
                        <h4>
                                    0.407
                        </h4>
            </div>
            <div>
                        <h3>
                                    Differential
                        </h3>
                        <h4
                                    className="compare-total"
                        >
                                    0.658
                        </h4>
            </div>
            <div>
                        <h3>
                                    AKRON R-1
                        </h3>
                        <h4>
                                    0.619
                        </h4>
            </div>
</article>,
          <Card
            handleCompareCards={[Function]}
            schoolData={
                        Object {
                                    "data": Object {
                                      "2004": 0,
                                      "2005": 0,
                                      "2006": 0,
                                      "2007": 0,
                                      "2008": 0.844,
                                      "2009": 0.96,
                                      "2010": 1,
                                      "2011": 1,
                                      "2012": 1,
                                      "2013": 1,
                                      "2014": 1,
                                    },
                                    "dataType": "Percent",
                                    "location": "AKRON R-1",
                                  }
            }
            size="large"
/>,
          <button
            onClick={[Function]}
>
            Reset
</button>,
        ],
        "className": "displayAll",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleCompareCards": [Function],
            "schoolData": Object {
              "data": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
              "dataType": "Percent",
              "location": "ACADEMY 20",
            },
            "size": "large",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h3>
                                ACADEMY 20
                </h3>
                <h4>
                                0.407
                </h4>
</div>,
              <div>
                <h3>
                                Differential
                </h3>
                <h4
                                className="compare-total"
                >
                                0.658
                </h4>
</div>,
              <div>
                <h3>
                                AKRON R-1
                </h3>
                <h4>
                                0.619
                </h4>
</div>,
            ],
            "className": "comparison",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    ACADEMY 20
</h3>,
                  <h4>
                    0.407
</h4>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ACADEMY 20",
                  },
                  "ref": null,
                  "rendered": "ACADEMY 20",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0.407,
                  },
                  "ref": null,
                  "rendered": 0.407,
                  "type": "h4",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Differential
</h3>,
                  <h4
                    className="compare-total"
>
                    0.658
</h4>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Differential",
                  },
                  "ref": null,
                  "rendered": "Differential",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0.658,
                    "className": "compare-total",
                  },
                  "ref": null,
                  "rendered": 0.658,
                  "type": "h4",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    AKRON R-1
</h3>,
                  <h4>
                    0.619
</h4>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "AKRON R-1",
                  },
                  "ref": null,
                  "rendered": "AKRON R-1",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0.619,
                  },
                  "ref": null,
                  "rendered": 0.619,
                  "type": "h4",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleCompareCards": [Function],
            "schoolData": Object {
              "data": Object {
                "2004": 0,
                "2005": 0,
                "2006": 0,
                "2007": 0,
                "2008": 0.844,
                "2009": 0.96,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "dataType": "Percent",
              "location": "AKRON R-1",
            },
            "size": "large",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reset",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
